astropylibrarian/algolia/records.py

```diff
@@ -1,110 +1,314 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
-"""Record types that reflect Algolia search records.
-"""
+"""Record types that reflect Algolia search records."""
 
-__all__ = ["TutorialSectionRecord"]
+from __future__ import annotations
+
+__all__ = ["ContentType", "AlgoliaRecord", "TutorialRecord", "GuideRecord"]
 
 import datetime
+import json
 from base64 import b64encode
-from dataclasses import dataclass, field
-from typing import TYPE_CHECKING, Any, Dict, List
+from enum import Enum
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
 from urllib.parse import urlparse, urlunparse
 
-from astropylibrarian.keywords import KeywordDb
+from pydantic import UUID4, BaseModel, Field, HttpUrl, validator
 
 if TYPE_CHECKING:
+    from astropylibrarian.keywords import KeywordDb
     from astropylibrarian.reducers.tutorial import ReducedTutorial
     from astropylibrarian.reducers.utils import Section
+    from astropylibrarian.reduers.jupyterbook import (
+        JupyterBookMetadata,
+        JupyterBookPage,
+    )
+
+
+class ContentType(str, Enum):
+    """Learn Astropy content types."""
+
+    tutorial = "tutorial"
+
+    guide = "guide"
+
+    example = "example"
+
+    documentation = "documentation"
+
+
+class AlgoliaRecord(BaseModel):
+    """A Pydantic model for an Learn Astropy record in Algolia."""
```

Comment:
```suggestion
    """A Pydantic model for a Learn Astropy record in Algolia."""
```
Fitting for this line that this is a *very* pedantic suggestion 😅